--- a/include/linux/types.h	2026-09-28 20:09:28.024791511 +0000
+++ b/include/linux/types.h	2026-09-28 20:09:28.025875205 +0000
@@ -3,6 +3,11 @@
 #include <asm/types.h>
 #ifndef __ASSEMBLY__
 #include <linux/posix_types.h>
+
+#ifdef __SIZEOF_INT128__
+typedef __signed__ __int128 __s128 __attribute__((aligned(16)));
+typedef unsigned __int128 __u128 __attribute__((aligned(16)));
+#endif
 /*
  * Below are truly Linux-specific types that should never collide with
  * any application/library that wants linux/types.h.
